port/esp32/adapter/src/ipadapter.c

```diff
@@ -1568,6 +1568,36 @@ oc_connectivity_init(size_t device, oc_connectivity_ports_t ports)
   return 0;
 }
 
+static void
+signal_event_thread(ip_context_t *dev)
+{
+  ssize_t result;
+  do {
+    result = write(dev->wakeup_pipe[1], "\n", 1);
+  } while (result == -1 && errno == EINTR);
+
+  if (result == -1) {
+    if (errno != ENOSPC) {
+      OC_WRN("Failed to write to wakeup pipe: %s", strerror(errno));
```

Comment:
func `strerror` is not always available - could you look how errno is solved in other parts in the file

Reply:
Fixed, the error code is logged instead of a string